src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs

```diff
@@ -795,7 +805,7 @@ void BuildDependencies(XmlNodeList dependencies)
                 return dependencyDetails;
             }
 
-            return dependencyDetails.Where(d => !d.Pinned);
+            return dependencyDetails.Where(d => !d.Pinned && !string.IsNullOrEmpty(d.RepoUri));
```

Comment:
I don't think this is the right way to go.  It fixes the symptom, but this should be dealt with in the caller.

Reply:
If we move this to the called this would still be prone to errors. Are there scenarios where we still want an empty repo?

Reply:
No, but we can handle it in places that are processing the dependency details (e.g. the dependency graph builder).

Reply:
Hmmm, so we'd need to filter those out on each call rather than a single place... Don't know if I like that much. I see your point of letting the user handle things as they need/want but since this is not a valid thing to begin with I'd rather filter the source

Reply:
However, this is not a valid state to be in.  So filtering out just hides the errors

Reply:
Agree and given that it is not valid I'll throw instead of filtering.